### PR TITLE
feat(Message): add crosspostable property

### PR DESCRIPTION
**Pull request by HarmoGlace**
**Please describe the changes this PR makes and why it should be merged:**

This PR adds the Message#crosspostable property, to know if a message can be crossposted.

**Status**

- [X] Code changes have been tested against the Discord API, or there are no code changes
- [X] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/Message.js [hunk 1]
```diff
@@ -418,6 +418,22 @@ class Message extends Base {
...
+   * Whether the message is crosspostable by the client user
+   * @type {boolean}
+   * @readonly
+   */
+  get crosspostable() {
+    return (
+      this.channel.type === 'news' &&
+      !this.flags.has(MessageFlags.FLAGS.CROSSPOSTED) &&
+      !this.flags.has(MessageFlags.FLAGS.IS_CROSSPOST) &&
+      this.author.id === this.client.user.id &&
```
Comment (NotSugden) [inline src/structures/Message.js:431]: can't you publish other users messages? or are bots restricted to their own?

Comment (wasdennnoch) [reply src/structures/Message.js:431]: You can indeed crosspost someone else's message if you have `MANAGE_MESSAGES`. [docs](https://discord.com/developers/docs/resources/channel#crosspost-message)

Comment (HarmoGlace) [reply src/structures/Message.js:431]: Indeed, I'll update it to support that

**Review by kyranet — APPROVED** [review]

File: src/structures/Message.js [hunk 2]
```diff
@@ -418,6 +418,23 @@ class Message extends Base {
...
+   */
+  get crosspostable() {
+    return (
+      this.channel.type === 'news' &&
+      !this.flags.has(MessageFlags.FLAGS.CROSSPOSTED) &&
+      !this.flags.has(MessageFlags.FLAGS.IS_CROSSPOST) &&
+      this.type === 'DEFAULT' &&
+      (this.author.id === this.client.user.id
+        ? this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.SEND_MESSAGES)
+        : this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES))
```
Comment (vladfrangu) [inline src/structures/Message.js:434]: You need both permissions if you're not the author
Also, VIEW_CHANNEL

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/structures/Message.js [hunk 3]
```diff
@@ -418,6 +418,25 @@ class Message extends Base {
...
+  /**
+   * Whether the message is crosspostable by the client user
+   * @type {boolean}
+   * @readonly
+   */
+  get crosspostable() {
+    return (
+      this.channel.type === 'news' &&
+      !this.flags.has(MessageFlags.FLAGS.CROSSPOSTED) &&
+      !this.flags.has(MessageFlags.FLAGS.IS_CROSSPOST) &&
```
[the remark this answers is not in the document]
Comment (HarmoGlace) [reply src/structures/Message.js]: Yes, I changed it

**Review by vladfrangu — APPROVED** [review]

File: src/structures/Message.js [hunk 4]
```diff
@@ -418,6 +418,24 @@ class Message extends Base {
...
 
+  /**
+   * Whether the message is crosspostable by the client user
+   * @type {boolean}
+   * @readonly
+   */
+  get crosspostable() {
+    return (
+      this.channel.type === 'news' &&
+      !this.flags.any([MessageFlags.FLAGS.CROSSPOSTED, MessageFlags.FLAGS.IS_CROSSPOST]) &&
```
Comment (advaith1) [inline src/structures/Message.js:429]: crossposts can be crossposted
```suggestion
      !this.flags.has(MessageFlags.FLAGS.CROSSPOSTED) &&
```

File: src/structures/Message.js [hunk 5]
```diff
@@ -418,6 +418,24 @@ class Message extends Base {
...
+  get crosspostable() {
+    return (
+      this.channel.type === 'news' &&
+      !this.flags.has(MessageFlags.FLAGS.CROSSPOSTED) &&
+      this.type === 'DEFAULT' &&
+      this.channel.viewable &&
+      this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.SEND_MESSAGES) &&
+      (this.author.id !== this.client.user.id
+        ? this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES)
+        : true)
```
Comment (TeeSeal) [inline src/structures/Message.js:435]:
```suggestion
      (
        this.author.id === this.client.user.id ||
        this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES)
      )
```

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**